docs/02_database/05_database-interface/01_entitydb.md

```diff
@@ -21,7 +21,55 @@ entity. It supports read operations for views and tables and write operations fo
 
 The entity db differs from the generated repositories in that it can handle any table and most view entities. It differs from `RxDb` in that all operations are type-safe.
 
-The entity db is available in the kotlin Event Handler. It can be injected in Kotlin using `AsyncEntityDb` and in Java using `RxEntityDb`.
+The entity db is available in the kotlin Event Handler. 
+
+
+It can be injected in Kotlin using `AsyncEntityDb` and in Java using `RxEntityDb`.
+
+There are three versions available of the entity db:
+* `AsyncEntityDb` - this API support kotlin coroutines 
+* `SyncEntityDb` - this API is blocking 
+* `RxEntityDb` - this API supports RxJava
+
+:::important
+
+The `SyncEntityDb` API was introduced in version 7.0 of the Genesis platform.
+
+:::
+
+## Which version to use?
+
+The version to use depends on the context. It depends on whether you are writing code in kotlin or java, and also 
+whether there is a need to write asynchronous (async) or synchronous (blocking) code. 
+
+|        | Asynchronous    | Blocking       |
+|--------|-----------------|----------------|
+| Kotlin | `AsyncEntityDb` | `SyncEntityDb` |
+| Java   | `RxEntityDb`    | `SyncEntityDb` |
+
+The difference between blocking and async code is that if the code needs to wait on the database, in
+blocking code, the thread will block until the database responds. In async code, the thread will not block,
+rather it will resume once the database responds. The general rule is that blocking code is easier to write, and 
+async code is more performant. Another thing to bear in mind is that within a single code path, you should not 
+mix blocking and async code in a single code path. Doing so can lead to unexpected deadlocks.
+
+We have to flavours of async API, kotlin coroutines and RxJava. 
+
+Coroutines are a language feature of kotlin, and allow you to write async code in a blocking style. Functions are 
```

Comment:
Include a link to Kotlin's doc on co-routine.

Reply:
moved this to the types of api docs where we have that already